test/Jobs.js

```diff
@@ -1,33 +1,151 @@
-    publicJob = { ...TestData.PublicJob };
-    done();
+let datasetPid1 = null,
+  datasetPid2 = null,
+  datasetPid3 = null,
+  jobId1 = null,
+  encodedJobId1 = null,
+  jobId2 = null,
+  encodedJobId2 = null,
+  jobId3 = null,
+  encodedJobId3 = null,
+  jobId4 = null,
+  encodedJobId4 = null,
+  jobId5 = null,
+  encodedJobId5 = null,
+  jobId6 = null,
+  encodedJobId6 = null,
+
+  jobIdGroup1 = null,
+  encodedJobIdGroup1 = null,
+  jobIdGroup2 = null,
+  encodedJobIdGroup2 = null,
+  jobIdGroup3 = null,
+  encodedJobIdGroup3 = null,
+  jobIdGroup4 = null,
+  encodedJobIdGroup4 = null,
+  jobIdGroup5 = null,
+  encodedJobIdGroup5 = null,  
+  jobIdGroup6 = null,
+  encodedJobIdGroup6 = null,
+
+  jobIdUser1 = null,
+  encodedJobIdUser1 = null,
+  jobIdUser2 = null,
+  encodedJobIdUser2 = null,
+  jobIdUser3 = null,
+  encodedJobIdUser3 = null,
+  jobIdUser4 = null,
+  encodedJobIdUser4 = null,
+  jobIdUser5 = null,
+  encodedJobIdUser5 = null,  
+  jobIdUser6 = null,
+  encodedJobIdUser6 = null,
+
+  jobIdUserSpec1 = null,
+  encodedJobIdUserSpec1 = null,
+  jobIdUserSpec2 = null,
+  encodedJobIdUserSpec2 = null,
+  jobIdUserSpec3 = null,
+  encodedJobIdUserSpec3 = null,
+  jobIdUserSpec4 = null,
+  encodedJobIdUserSpec4 = null,
+  jobIdUserSpec5 = null,
+  encodedJobIdUserSpec5 = null,  
+  jobIdUserSpec6 = null,
+  encodedJobIdUserSpec6 = null,
+  jobIdUserSpec7 = null, 
+  encodedJobIdUserSpec7 = null,
+
+  jobIdGroupSpec1 = null,
+  encodedJobIdGroupSpec1 = null,
+  jobIdGroupSpec2 = null,
+  encodedJobIdGroupSpec2 = null,
+  jobIdGroupSpec3 = null,
+  encodedJobIdGroupSpec3 = null,
+  jobIdGroupSpec4 = null,
+  encodedJobIdGroupSpec4 = null,
+  jobIdGroupSpec5 = null,
+  encodedJobIdGroupSpec5 = null,  
+  jobIdGroupSpec6 = null,
+  encodedJobIdGroupSpec6 = null,
+  jobIdGroupSpec7 = null, 
+  encodedJobIdGroupSpec7 = null,
+  jobIdGroupSpec8 = null,
+  encodedJobIdGroupSpec8 = null;
+
+
+const dataset1 = {
+  ...TestData.RawCorrect,
+  isPublished: true,
+  ownerGroup: "group1",
+  accessGroups: ["group5"],
+};
+
+const dataset2 = {
+  ...TestData.RawCorrect,
+  isPublished: false,
+  ownerGroup: "group2",
+  accessGroups: [],
+};
+
+const dataset3 = {
+  ...TestData.RawCorrect,
+  isPublished: false,
+  ownerGroup: "group5",
+  accessGroups: ["group1"],
+};
+const jobAll = {
+  ...TestData.Job, 
+  type: "all_access",
+};
+const jobDatasetPublic = {
+  ...TestData.Job, 
+  type: "public_access",
+}
+const jobAuthenticated = {
+  ...TestData.Job, 
+  type: "authenticated_access"
+};
+const jobDatasetAccess = {
+  ...TestData.Job, 
+  type: "dataset_access"
+};
+const jobDatasetOwner = {
+  ...TestData.Job, 
+  type: "owner_access"
+};
+const jobUser51 = {
+  ...TestData.Job, 
+  type: "user_access"
+};
+const jobGroup5 = {
+  ...TestData.Job, 
+  type: "group_access"
+};
+
+describe("1100: Jobs: Test New Job Model", () => {
+  before(() => {
+    db.collection("Dataset").deleteMany({});
+    db.collection("Job").deleteMany({});
```

Comment:
I have been trying to figure out why this part is not working. We expect that it would empty the `Dataset` and `Job` collections, so that we can start the tests with an empty database, however this does not happen. 

One idea to solve this would be to prune the `mongodb volume` before starting the test, but I do not know whether this is the right approach. 

In my effort to understand what these two lines of code are doing, I came across the file `pretest.js` file, which contains the following:
```
const uri = "mongodb://localhost:27017/scicat";
const client = new MongoClient(uri);
global.db = client.db("scicat");
```
So now my question is why are we trying to connect to `scicat` db? Shouldn't it be `test_be_next`? Or perhaps there is something that I don't quite understand. Could someone please this to me? Perhaps @sofyalaski or @Junjiequan ?

Reply:
It's my mistake. the db name should be customizable. I believe there's variable named test_db or sth, we can change to that instead hardcode here.